core/controller/src/main/scala/whisk/core/controller/WebActions.scala

```diff
@@ -483,7 +483,24 @@ trait WhiskWebActionsApi extends Directives with ValidateRequestSize with PostAc
               provide(fullyQualifiedActionName(actionName)) { fullActionName =>
                 onComplete(verifyWebAction(fullActionName, onBehalfOf.isDefined)) {
                   case Success((actionOwnerIdentity, action)) =>
-                    if (!action.annotations.getAs[Boolean]("web-custom-options").exists(identity)) {
+                    val requireWebAuthIsBool = (action.annotations.getAs[Boolean]("require-whisk-auth") != None)
+                    val annotationRequireWebAuthIsIntOrString = ((action.annotations
+                      .getAs[String]("require-whisk-auth") != None) || (action.annotations
+                      .getAs[Int]("require-whisk-auth") != None))
+                    val annotationRequireWebAuth = (action.annotations
+                      .getAs[Int]("require-whisk-auth")
+                      .getOrElse(action.annotations.getAs[String]("require-whisk-auth").getOrElse("")))
+                      .toString
```

Comment:
`Option` has another combinator called `orElse` which takes another `Option` if the first one was `None`. In your case this would look like this:

```scala
val annotationRequireWebAuth: Option[String] = action.annotations
  .getAs[Int]("require-whisk-auth").map(_.toString)
  .orElse(action.annotations.getAs[String]("require-whisk-auth"))
```

If you leave out the default of `""` you can also use this value to determine if its Int or String by using `isDefined`. That makes `annotationRequireWebAuthIsIntOrString` obsolete.

Reply:
i like it :+1: